tutorials/polar_component/scripts/polar_pipe.py

```diff
@@ -0,0 +1,106 @@
+from spacy import Language
+from spacy.tokens import Doc
+from spacy.pipeline import Pipe
+from spacy.vocab import Vocab
+
+from thinc.types import Floats1d, Floats2d
+from thinc.api import Model, CosineDistance, get_ops
+
+from dataclasses import dataclass
+
+Doc.set_extension("poles", default={}, force=True)
+
+
+@dataclass
+class Axis:
+    """An invididual semantic axis."""
+
+    neg: str
+    pos: str
+    vector: Floats1d
+
+    def get_key(self, sep="-"):
+        return f"{self.neg}{sep}{self.pos}"
+
+
+@Language.factory(
+    "polar",
+    requires=["doc.vector"],
+    default_config={},
+    default_score_weights={},  # ???
```

Comment:
Why the comment here?

Reply:
Actually I forget... I guess it can just be removed.